moses/moses/main/populate_atomspace.cc

```diff
@@ -0,0 +1,70 @@
+#include <moses/data/table/table.h>
+#include <moses/data/table/table_io.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <moses/utils/value_key.h>
+#include <opencog/atoms/base/Link.h>
+#include "populate_atomspace.h"
+
+namespace opencog {
+    namespace moses {
+
+        void populate(AtomSpace *as, ITable &itable) {
+
+            std::vector<multi_type_seq>::const_iterator it, end;
+            int col_size = itable.get_types().size();
+            for (int i = 0; i < col_size; i++) {
+                id::type_node col_type = itable.get_types().at(i);
+
+                int row_size = itable.size();
+
+                switch (col_type) {
+                    case id::boolean_type: {
+                        std::vector<ProtoAtomPtr> col_values = {};
+                        Handle feature = createNode(SCHEMA_NODE,"f"+ std::to_string(i+1));
+                        for (int j = 0; j < row_size; j++) {
+                            // for each element the ith column and the jth row
+                            // change the vertex to bool
+                            bool col_data = vertex_to_bool(itable.get_column_data(itable.get_labels().at(i)).at(j));
```

Comment:
Because extracting a column has some cost, it would be better if you store the column outside of the loop over `j`. That way you only need to access it rather than recreate it. So I mean, you'd have before the `for (int j = 0; ...` loop
```c++
vertex_seq col = get_column_data(i);
```
then within the `j` loop
```c++
bool col_data = vertex_to_bool(col.at(j));
```